### DISCUSSION TITLE
Log in with Linkedin throws an error: "Unsupported provider: provider is not enabled" even after all settings are good

### DISCUSSION THREAD

**Discussion by hurricanehunter0702** [supabase/supabase#21076] [Questions]
Hi, team. I appreciate for your efforts.

Login with "Google" and "Facebook" are fine.
The problem is "Linkedin" provider is enabled in the Supabase authentication settings and I entered the correct credentials.
But it is still throwing the error "Unsupported provider: Provider is not enabled".

I followed this guide to enable Linkedin Provider:
https://supabase.com/docs/guides/auth/social-login/auth-linkedin

Please help me on this matter. Could anyone please let me know what I'm missing?

**Comment by GaryAustin1** [comment]
Just want to make sure you are using the OIDC version.  
![image](https://github.com/supabase/supabase/assets/54564956/da0653da-5f48-49ff-807b-7dc4a75a63dc)

**Reply by hurricanehunter0702** [reply]
Yeah, I am using it.
![image](https://github.com/supabase/supabase/assets/124948717/56b6f4b8-d759-4a45-9a3a-ec5fe0a36a4c)

**Reply by hurricanehunter0702** [reply]
And I'm using @supabase/auth-ui-react.
I'm not sure if it causes the problem. It's just for a reference.

**Reply by GaryAustin1** [reply]
How are you specifying linkedin there.  My guess is it is trying to go to the old linked in which is not longer supported.

**Reply by GaryAustin1** [reply]
![image](https://github.com/supabase/supabase/assets/54564956/b9a8dba1-6bc1-40db-be81-04bb7ac0a326)

**Reply by mkultracreative** [reply]
What do you mean “the old LinkedIn”?

> ![image](https://private-user-images.githubusercontent.com/54564956/303108795-b9a8dba1-6bc1-40db-be81-04bb7ac0a326.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyNzMzMjcsIm5iZiI6MTcyNjI3MzAyNywicGF0aCI6Ii81NDU2NDk1Ni8zMDMxMDg3OTUtYjlhOGRiYTEtNmJjMS00MGRiLWJlODEtMDRiYjdhYzBhMzI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE0VDAwMTcwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmMWRhM2M3MGRlMTVmODhjZDMzMmExNjdiZmZkOTkzZDg5ZGY2ZDliOWNkYmMwMDY3MTFhYzNkOTc5NTNjYzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9AyboAcxseTgvTYBY-ns8moAw8sA7FfaPQK2kpynOpI)

**Comment by mdrashedbinkuddus** [comment]
What about Flutter? Getting the same issue

`{
  "code": 400,
  "error_code": "validation_failed",
  "msg": "Unsupported provider: provider is not enabled"
}`

**Reply by dinisafonso410-beep** [reply]
Agora o que posso fazer

**Reply by mdrashedbinkuddus** [reply]
Use latest supabase package

**Comment by mkultracreative** [comment]
I’m getting the same error!!! This is super annoying especially after the docs explain so little in supabase about how to actually implement this or which option exactly to use. It even at one point says to use “this PWK method” then basically gives totally diff code without explaining why for the same sign in function but using a different method then literally one paragraph later says “it’s not supported migrate over to oidc” which is exactly the page you’re on. Then says to use the “general oAuth method” in general server settings. Even though the server options for this auth are also listed in the same page. Docs are crap.

**Comment by AndonMitev** [comment]
getting the same but for github...

**Comment by mkultracreative** [comment]
# Found the mole
Okay so this is the worst part about this error: 

- ❌ Never once does any error via linting, cli, client facing DOM, or the printed on screen error, catch the actual problem.

- ⚠️ This tells me the bug is actually in the error itself. 
- ☢️ But it could also point to a deeper core issue in the provider session context where it's technically catching the correct error, but the code that handles this stuff is out of order somehow. I _can't_ know because the entire debugging for this is limited to the limited amount of telemetry for this method

- [x] I did [fix the issue](#solutionish). But don't get too excited yet...it's near impossible to reproduce again on another project after literally cloning this.

# Solutionish
- Somehow, it's the redirect url specifically in [Supabase Dashboard Settings](https://supabase.com/dashboard/project/_/auth/url-configuration)
> ✋ Hold your horses. Of course I assumed you tried this. I did too. But did you do [this](#actual-server)?

### Actual server

1. Forget whether you're local or prod or dev env, forget all of that. Just open the actual link that's launched:
> If it's Vercel, you're not going to see `localhost:3000-8100` or whatever server you're setting in routes. You'll see the domain actually hosting the preview.
2. Copy that exact URL with the `https://[actual-domain]/api/auth/providers/linkedin` or for codespaces, `https://scaling-robot-somelongassstring/...` or Supabase, `https://supabase.com/dashboard/project/_/auth/v1/callback` -- url [located here](https://supabase.com/dashboard/project/_/auth/providers). for Vercel, you get the idea
3. Here's the important thing though, you have to use the _full_ callback url in 3 places, _without any wildcards regardless of whether the provider supports it_.
---
#### Supabase
- Allowed URL's
- Site URL

#### Provider: LinkedIn
- [OAuth 2.0 Settings](https://www.linkedin.com/developers/apps/_/auth)

#### If using Auth.js
- The callback will resolve to the following so you must add this URL to any allowed lists and default redirect `https://[origin]/api/auth/callback/providers/linkedin`


If that doesn't work, I'm telling you, it's not a code base issue. Your code is fine. It's a provider config setting issue. 

At one point I even updated my client secret and added the new one in the Supa Provider settings. This may have had an impact. 

Good luck!

**Reply by mkultracreative** [reply]
Also check to make sure you’re using the correct prefix notation in .env secrets. For example:

#### Supabase

```
NEXT_PUBLIC_SUPABASE_URL=your_supabase_project_url

NEXT_PUBLIC_SUPABASE_ANON_KEY=your_supabase_anon_key
```

#### [Auth.js](https://auth.js) check the [provider page here](https://authjs.dev/getting-started/authentication/oauth?provider=linkedin)

```
AUTH_LINKEDIN_ID={CLIENT_ID}
AUTH_LINKEDIN_SECRET={CLIENT_SECRET}

```
Lastly, for Supabase this may fix it outright — add a config.toml file as stated [here](https://supabase.com/docs/guides/auth/social-login/auth-linkedin?queryGroups=environment&environment=client):

```
[auth.external.linkedin_oidc]
enabled = true
client_id = ...
secret = ...
```

**Comment by mirkenstein** [comment]
Got it working. Thanks @GaryAustin1  for pointing this out.
Use provider `linkedin_oidc` not `linkedin` and for the scopes: use 
 ` openid profile email` not `r_liteprofile r_emailaddress'`
See also this https://community.auth0.com/t/the-linkedin-login-is-deprecated-updating-to-the-new-scopes-is-necessary/113696 
```js
  const handleLinkedInSignIn = async () => {
    try {
      const { data, error } = await supabase.auth.signInWithOAuth({
        provider: 'linkedin_oidc',
        options: {
          scopes: 'openid profile email',
          redirectTo: `${window.location.origin}/`
        }
      });
```
